machine/quanta/quanta_common_p2020/INSTALL

```diff
@@ -0,0 +1,165 @@
+   1xRJ-45 console port
+   1xmicro SD card
+   1xUSB 2.0 port
+
+  Special Feature
+=================
+ - ONIE recovery mechanism
+   Please refer to ``Recovering ONIE back to factory default``
+
+================================================
+Installing ONIE on Quanta Common P2020 CPU board
+================================================
+
+Cross-Compiling ONIE
+====================
+
+Change directories to ``build-config`` to compile ONIE.
+
+To compile ONIE first change directories to ``build-config`` and then
+type ``"make MACHINEROOT=../machine/quanta/ MACHINE=quanta_common_p2020 all"``.
+For example:
+
+  $ cd build-config
+  $ make -j4 MACHINEROOT=../machine/quanta/ MACHINE=quanta_common_p2020 all
+
+When complete, the ONIE binaries are located in
+``build/images``::
+
+  -rw-r--r-- 4849664 Aug 12 15:39 onie-quanta_common_p2020-r0.bin
+
+Or, change directories to ``machine/quanta/quanta_common_p2020`` to compile ONIE,
+use ``build.sh`` to compile ONIE. For example:
+
+  $ cd machine/quanta/quanta_common_p2020
+  $ ./build.sh all
+
+To compile demo installer.
+
+  $ cd machine/quanta/quanta_common_p2020
+  $ ./build.sh demo
+
+To compile html and pdf documents.
+
+  $ cd machine/quanta/quanta_common_p2020
+  $ ./build.sh docs
+
+Installing the ONIE binaries
+============================
+
+A single binary image needs to be installed on the NOR flash.
+
+Image 1 -- ONIE kernel and u-boot.
+  name:  onie-quanta_common_p2020-r0.bin
+
+Step 1 -- Put the ONIE file on a TFTP server
+---------------------------------------------
+
+The following directions assume the files are on the root of the TFTP
+server.
+
+Step 2 -- Install image1 (onie-quanta_common_p2020-r0.bin)
+-----------------------------------------------
+
+Copying the image down using TFTP and flash to the NOR flash::
+
+  => setenv start 0xefb60000
+  => setenv sz.b 0x4a0000
+  => tftp onie-quanta_common_p2020-r0.bin
+  => protect off $start +${sz.b} && erase $start +${sz.b}
+  => cp.b $fileaddr $start ${sz.b} && protect on $start +${sz.b}
+
+Step 3 -- Configure Serial Console
+----------------------------------
+
+ONIE defaults the serial console baud rate to 115200.  You may need to
+adjust your terminal settings.
+
+Step 4 -- Restart The System
+----------------------------
+
+You can interrupt the boot process by pressing any key during the
+count down::
+
+  => reset
+
+Step 5 -- Optional
+------------------
+
+By default the system will start the ONIE in *install* mode.  After
+booting press the [Enter] key to active the console.
+
+Alternatively you can start the system in *rescue* mode so you can
+login and look around the ONIE.  To start the rescue mode type this::
+
+  => setenv reboot_cmd rescue
+  => boot
+
+Or, in another way
+
+  => run onie_rescue
+
+
+Updating the ONIE binaries
+==========================
+
+An easier way to update ONIE within u-boot
+
+  => setenv ipaddr 192.168.2.1
+  => copy -o tftp://192.168.2.101/onie-quanta_common_p2020-r0.bin
```

Comment:
Yes, it is taken as a macro command of all the update onie steps.
Simple, and also checks the size of download file.